### PR TITLE
MOTH - verify MOT certificate printing

### PR DESCRIPTION
**Pull request by adrianz-kainos**
Please do not merge this yet. We need to enable the proper feature toggle first:
https://jira.i-env.net/browse/BL-9789

### CODE REVIEW

**Comment by adrianz-kainos** [comment]
Please do not merge this yet. We need to enable the proper feature toggle first:
https://jira.i-env.net/browse/BL-9789

File: src/main/resources/features/moth/54 - Print MOT certificate.feature [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+Feature: 54 - Check whether MOT certificates can be downloaded after entering the V5C number
+
+  Scenario: A MOTH user searches for a vehicle with an MOT test and downloads the MOT certificate
+    Given I browse to /
+    And I enter "CGSENOK" in the registration field
+    When I press the "Continue" button
+
+    And I click the accordion section with the id "mot-history-description"
+    And I click the first "View test certificate" text
+    And The page contains "Enter latest V5C number"
```
Comment (ryszardm-kainos) [inline src/main/resources/features/moth/54 - Print MOT certificate.feature]: Scenarios should follow the Given/When/Then pattern - Its hard to understand here what is the path taken and what is the expected result here

**Comment by ibrahimjassat** [comment]
Needs a green run before its merged

File: src/main/resources/features/moth/54 - Print MOT certificate.feature [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+    And I press the first "Show test certificate" button
+    And The page contains "Check that the V5C number you entered is correct"
+
+    And I enter "12312312312" in the field with id "v5c-print-cert-input"
+    And I press the first "Show test certificate" button
+
+    And The page contains "View certificate"
+    And I check the "Return to MOT test history" button is enabled
+    And I click the button with id "cert-download-link"
+
```
Comment (ibrahimjassat) [inline src/main/resources/features/moth/54 - Print MOT certificate.feature]: Can include a check if the PDF of the certificate looks correct here

File: src/main/resources/features/moth/54 - Print MOT certificate.feature [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+    And I enter "CGSENOK" in the registration field
+    When I press the "Continue" button
+
+    And I click the accordion section with the id "mot-history-description"
+    And I click the first "View test certificate" text
+    And The page contains "Enter latest V5C number"
+
+    And I enter "00000000000" in the field with id "v5c-print-cert-input"
+    And I press the first "Show test certificate" button
+    And The page contains "Check that the V5C number you entered is correct"
```
Comment (ibrahimjassat) [inline src/main/resources/features/moth/54 - Print MOT certificate.feature]: Would split this into two different tests one for happy path and one for the unhappy path

File: src/main/java/uk/gov/dvsa/mot/fixtures/generic/WebStepDefinitions.java [hunk 4]
```diff
@@ -33,8 +33,8 @@ public WebStepDefinitions(WebDriverWrapper driverWrapper) {
         When("^I press the \"([^\"]+)\" button$", (String buttonText) ->
                 driverWrapper.pressButton(buttonText));
 
-        When("^I press the first \\{([^\\}]+)\\} button$", (String buttonText) ->
-                driverWrapper.pressFirstButton(driverWrapper.getData(buttonText)));
+        When("^I press the first \"([^\"]+)\" button$", (String buttonText) ->
```
Comment (karlbeckettbjss) [inline src/main/java/uk/gov/dvsa/mot/fixtures/generic/WebStepDefinitions.java:36]: Changing this step will cause other tests to fail. Is there a reason why it changed?

Comment (adrianz-kainos) [reply src/main/java/uk/gov/dvsa/mot/fixtures/generic/WebStepDefinitions.java:36]: My bad. I was unable to find any usages of this step. I can now see that it's used somewhere else. I will revert this change.

File: src/main/resources/features/moth/54 - Print MOT certificate.feature (hunk 2 above)
Comment (adrianz-kainos) [reply src/main/resources/features/moth/54 - Print MOT certificate.feature]: Checks added

File: src/main/java/uk/gov/dvsa/mot/fixtures/generic/WebStepDefinitions.java (hunk 4 above)
Comment (adrianz-kainos) [reply src/main/java/uk/gov/dvsa/mot/fixtures/generic/WebStepDefinitions.java:36]: I created a new step definition, the one mentioned above should remain intact.

File: src/main/resources/features/moth/54 - Print MOT certificate.feature (hunk 3 above)
Comment (adrianz-kainos) [reply src/main/resources/features/moth/54 - Print MOT certificate.feature]: Done

File: src/main/resources/features/moth/54 - Print MOT certificate.feature (hunk 1 above)
Comment (adrianz-kainos) [reply src/main/resources/features/moth/54 - Print MOT certificate.feature]: Done

**Comment by mrogowska** [comment]
Tests under @mothprint tag passed: https://jenkins2.motdev.org.uk/job/MOTH/job/MOTH_selenium/534/

File: src/main/resources/features/moth/55 - Print MOT certificate successfully.feature [hunk 5]
```diff
@@ -0,0 +1,41 @@
+@mothprint @mothint
+Feature: 54 - Check whether MOT certificates can be downloaded after entering the V5C number
```
Comment (ibrahimjassat) [inline src/main/resources/features/moth/55 - Print MOT certificate successfully.feature]: Feature number should be changed from 54 to 55 to match the file

**Review by ibrahimjassat — CHANGES_REQUESTED** [review]

File: src/main/java/uk/gov/dvsa/mot/fixtures/generic/WebStepDefinitions.java [hunk 6]
```diff
@@ -33,6 +33,9 @@ public WebStepDefinitions(WebDriverWrapper driverWrapper) {
         When("^I press the \"([^\"]+)\" button$", (String buttonText) ->
                 driverWrapper.pressButton(buttonText));
 
+        When("^I press the first \"([^\"]+)\" button$", (String buttonText) ->
```
Comment (karlbeckettbjss) [inline src/main/java/uk/gov/dvsa/mot/fixtures/generic/WebStepDefinitions.java:36]: Please add this new step definition to the docs

mot-automated-testsuite/docs/step-definitions.md

File: src/main/java/uk/gov/dvsa/mot/fixtures/mot/DocumentStepDefinitions.java [hunk 7]
```diff
@@ -35,10 +35,28 @@ public DocumentStepDefinitions(WebDriverWrapper driverWrapper) {
         And("^I click \"([^\"]+)\" and check the PDF contains:$",
                 (String link, DataTable table) -> assertTrue(pdfContainsData(link, table)));
 
+        And("^I click the certificate link and check the MOT certificate PDF contains:$",
```
Comment (karlbeckettbjss) [inline src/main/java/uk/gov/dvsa/mot/fixtures/mot/DocumentStepDefinitions.java:38]: Please add this new step definition to the docs

mot-automated-testsuite/docs/step-definitions.md

**Review by karlbeckettbjss — CHANGES_REQUESTED** [review]

File: src/main/resources/features/moth/55 - Print MOT certificate successfully.feature (hunk 5 above)
Comment (mrogowska) [reply src/main/resources/features/moth/55 - Print MOT certificate successfully.feature]: Number changed.

File: src/main/resources/features/moth/54 - Print MOT certificate with failures.feature [hunk 8]
```diff
@@ -0,0 +1,37 @@
+@mothprint @mothint
+Feature: 54 - Check whether MOT certificates can be downloaded after entering the V5C number
+
+  Scenario: A MOTH user searches for a vehicle with an MOT test and downloads the MOT certificate
```
Comment (karlbeckettbjss) [inline src/main/resources/features/moth/54 - Print MOT certificate with failures.feature]: The description for the feature and scenario doesn't match what the tests are doing, can this be changed.

File: src/main/java/uk/gov/dvsa/mot/fixtures/generic/WebStepDefinitions.java (hunk 6 above)
Comment (mrogowska) [reply src/main/java/uk/gov/dvsa/mot/fixtures/generic/WebStepDefinitions.java:36]: Added.

File: src/main/java/uk/gov/dvsa/mot/fixtures/mot/DocumentStepDefinitions.java (hunk 7 above)
Comment (mrogowska) [reply src/main/java/uk/gov/dvsa/mot/fixtures/mot/DocumentStepDefinitions.java:38]: Added.

File: src/main/resources/features/moth/54 - Print MOT certificate with failures.feature (hunk 8 above)
Comment (mrogowska) [reply src/main/resources/features/moth/54 - Print MOT certificate with failures.feature]: The feature description and scenario changed to match what is happening in the tests.

**Review by karlbeckettbjss — APPROVED** [review]

**Review by ibrahimjassat — APPROVED** [review]

**Pull request merged**